proposals/billboards-xformOp/README.md

```diff
@@ -0,0 +1,320 @@
+        uniform token[] xformOpOrder = ["xformOp:translate"]
+
+        double radius = 1.0
+        double height = 3.0
+        color3f[] primvars:displayColor = [(0.8, 0.8, 0.0)]
+    }
+}
+```
+In this example both `Cones` and `Cone2` defined their local transformations. The local
+transformation defined in `Cones` is a left-multiplied matrix of the local transformation
+of `Cone2`. In other words, transformation of `Cones` is a part of the transformation stack
+of `Cone2`. If you flatten the transformation stack, `Cone2` inherently defined an ordered
+set of `XformOp`s of [*Cones*:`xformOp:translate`, *Cones*:`xformOp:rotateXYZ`, `xformOp:translate`].
+
+Currently USD only support **simple** `xformOp`s. **Simple** means a transformation matrix
+can be computed solely on Op's attributes. In our previous example, the matrix of
+`xformOp:translate` can be computed with the authored offset vector ```(1.0, 0.0, 0.0)```.
+There are no external parameters required in this matrix computation. All `XformOp`s supported
+in USD right now have and only need one attribute to compute its local transformation.
+
+Billboards transformation is not simple. It is **compound** and **extrinsic**.
+
+### Examples of billboards transformation
+![Examples of billboards transformation](billboards-examples.png)
+
+This screenshot demonstrates some use cases of the billboards transformation:
+  - Callout: display screen facing texts at a fixed screen location with a fixed screen size.
+  - Pin: display a screen facing pin with a fixed scale.
+  - Axis tripod: display an axis tripod at a fixed screen location with a fixed screen size.
+
+### Billboards Transformation
+A billboards transformation places the renderable (or transformable in USD)
+in a predefined screen location, with a constant pixel-scaled size, or a
+locked facing alignment. A billboards transformation may consist of three types
+of transformations: affix the position, affix the size and align the front facing.
+Each of these transformations is defined as a `XformOp` in USD. XformOp `affixPosition`
+constructs the transformation that positions the transformable at an authored screen location.
+XformOp `affixScale` constructs the transformation that scales the renderable to an authored
+screen size. XformOp `align` constructs the transformation that maintains the orientation of
+transformable's front facing.
+
+`affixPosition`, `affixScale` and `align` are **compound** since they all require a set of
+attributes (instead of single one) to compute the transformation matrix. All three
+XformOp are **extrinsic** as they all request the current viewing states (such as
+the view matrix and the projection matrix) to compute the local transformation of the
+billboarded primitives (again, a transformable in USD).
+
+XformOps `affixPosition` and `affixScale` can specify the target position and size
+in a selected coordinate space:
+- `ndc`: coordinate is in the normalized device space ([-1.0, 1.0]x[-1.0, 1.0])
```

Comment:
I think it might be better to pick a different name for this.   Is there something simpler that a lay person might grok easier?

USD has generally avoided acronyms other than super common ones like IOR. I worry that NDC is too domain specific, and the acronym doesn't align with the words (NDC vs NDS). I don't think `device space` is necessarily easy to grok either. What would the `device` be in this scenario?

Reply:
`normalized device coordinate` is a standard term of art, and `ndc` is a standard "acronym of art". I don't think changing the term improves understandability, and I'm not convinced that spelling out ndc helps. Perhaps a a definition is in order. The reason I don't think spelling it out or changing the term helps is this: Artists are used to photoshop type coordinate systems with a 0,0 coordinate, typically upper left. a coordinate system centered in the viewport and extending in negative directions to indicate "up and left" is not at all what they're trained for. Leaving it as `ndc` means that there's an opportunity to prompt a discussion and some learning.

Just calling it `normalized` is super vague compared to the term of art. What's being normalized? Versus what origin and Euclidean metric? The only thing that would make sense to me is something like "viewport centered coordinate system with negative values being up and left and -1 to 1 are stretched to fit the whole viewport". It doesn't add anything over `ndc` ;)

Reply:
Yes, ndc is assumed to be xy on the near plane and view is the 3d position in eye space. Since you mentioned perspective transform, I assume you mean clip space?